src/value.rs

```diff
@@ -132,3 +141,33 @@ pub fn type_from_str(name: &str) -> Option<RbxValueType> {
         _ => None,
     }
 }
+
+struct Color3Value {
+    value: [f32; 3],
+}
+
+impl Color3Value {
+    pub fn new(value: [f32; 3]) -> Self {
+        Self { value }
+    }
+
+    fn meta_to_string<'lua>(&self, context: Context<'lua>) -> rlua::Result<rlua::Value<'lua>> {
+        format!("{}, {}, {}", self.value[0], self.value[1], self.value[2]).to_lua(context)
+    }
+}
+
+impl From<&Color3Value> for RbxValue {
+    fn from(color: &Color3Value) -> RbxValue {
+        RbxValue::Color3 {
+            value: color.value.clone(),
+        }
+    }
+}
```

Comment:
If we make our type `Copy`, we can drop all the references and `clone()` calls, which makes this a little easier.